### ISSUE TITLE
Why does ZFS forbid from setting bootfs pool property/removing top-level vdev with bootfs set "because GRUB does not understand it"

### ISSUE THREAD

**Issue by nabijaczleweli** [openzfs/zfs#11145]
### Ask your question!

```
root@tzpfmst:~# zpool remove tzpfmstoot vdb
cannot remove vdb: root pool can not have removed devices, because GRUB does not understand them
root@tzpfmst:~# zpool set bootfs=tzpfmstoot/root tzpfmstoot
cannot set property for 'tzpfmstoot': operation not supported on this type of pool
```

Why? Who cares what GRUB supports? Not being able to set this breaks `grub-mkconfig`'s root= autogeneration and initrds' root=zfs:auto, and using either EFI boot (most new installations) or legacy boot with /boot on ext2/FAT (trivally upgradable from old installations, and, indeed, the other half of my ZFS-on-root systems) works with this configuration, because GRUB doesn't touch ZFS pools at all.

### Which portion of the codebase does your question involve?

module/zfs/spa.c#spa_prop_validate() ~L620 (I think?):
```c
				/*
				 * Must be ZPL, and its property settings
				 * must be supported by GRUB (compression
				 * is not gzip, and large dnodes are not
				 * used).
				 */
```

lib/libzfs/libzfs_pool.c#zpool_vdev_remove() ~L3601:
```c
    if (!islog && !avail_spare && !l2cache && zpool_is_bootable(zhp)) {
		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
		    "root pool can not have removed devices, "
		    "because GRUB does not understand them"));
		return (zfs_error(hdl, EINVAL, msg));
	}
```

### Additional context

Can I just post a PR that strips these? Or is there an important reason to keep this?

**Comment by gmelikov** [comment]
Grub restrictions were removed in https://github.com/openzfs/zfs/pull/8627 , will be released in 2.0.

**Comment by nabijaczleweli** [comment]
oh, cool, good to know; thanks
